types/rational.go

```diff
@@ -61,6 +61,9 @@ func NewRatFromDecimal(decimalStr string) (f Rat, err Error) {
 		if len(str[0]) == 0 || len(str[1]) == 0 {
 			return f, ErrUnknownRequest("not a decimal string")
 		}
+		if len(str[1]) > prec {
+			str[1] = str[1][:prec]
+		}
```

Comment:
I think we should actually cause an error if somebody is sending in to much precision not just truncate it - we shouldn't fail silently should let user know